### PR TITLE
fix: remove erroneous check for the subject token field name for text credential source

### LINKED ISSUE #815: IdentityPoolCredential does not work for text formatted credential source
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/google-auth-library-java/issues
  - Check for answers on StackOverflow: http://stackoverflow.com/questions/tagged/google-cloud-platform

If you are still having issues, please include as much information as possible:

subject_token_field_name is only valid for `CredentialFormatType.JSON`, but is also run on `CredentialFormatType.TEXT` which cause it to throw the exception:

https://github.com/googleapis/google-auth-library-java/blob/5688837ca1fb3baab156af98126f0dbe8a668fd8/oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java#L139-L146

In the Python SDK, this line is if-checked to run only with JSON:

https://github.com/googleapis/google-auth-library-python/blob/3c3fbf40b07e090f2be7fac5b304dbf438b5cd6c/google/auth/identity_pool.py#L158-L165

#### Stack trace
```
java.lang.IllegalArgumentException","message":"When specifying a JSON credential type, the subject_token_field_name must be set.","stacktrace":"java.lang.IllegalArgumentException: When specifying a JSON credential type, the subject_token_field_name must be set.\n\tat com.google.auth.oauth2.IdentityPoolCredentials$IdentityPoolCredentialSource.<init>(IdentityPoolCredentials.java:144)\n\tat com.google.auth.oauth2.ExternalAccountCredentials.fromJson(ExternalAccountCredentials.java:319)\n\tat com.google.auth.oauth2.GoogleCredentials.fromStream(GoogleCredentials.java:170)\n\tat com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentialsUnsynchronized(DefaultCredentialsProvider.java:159)\n\tat com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentials(DefaultCredentialsProvider.java
```

### PR DESCRIPTION
**Pull request by lsirac**
Fixes #815

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java [hunk 1]
```diff
@@ -132,18 +132,21 @@
...
         String type = formatMap.get("type");
-        if (!"text".equals(type) && !"json".equals(type)) {
-          throw new IllegalArgumentException(
-              String.format("Invalid credential source format type: %s.", type));
-        }
-        credentialFormatType =
-            type.equals("text") ? CredentialFormatType.TEXT : CredentialFormatType.JSON;
 
-        if (!formatMap.containsKey("subject_token_field_name")) {
+        if ("json".equals(type)) {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java:136]: why not using CredentialFormatType.JSON?

Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java:136]: type.toLowerCase()?

File: oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java [hunk 2]
```diff
@@ -132,18 +132,21 @@
...
+          // For JSON, the subject_token field name must be provided.
+          if (!formatMap.containsKey("subject_token_field_name")) {
+            throw new IllegalArgumentException(
+                "When specifying a JSON credential type, the subject_token_field_name must be set.");
+          }
+          credentialFormatType = CredentialFormatType.JSON;
+          subjectTokenFieldName = formatMap.get("subject_token_field_name");
+        } else if ("text".equals(type)) {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java:144]: similarly toLowerCase ?

File: oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java [hunk 3]
```diff
@@ -415,6 +412,49 @@ void refreshAccessToken_workforceWithServiceAccountImpersonation() throws IOExce
...
+    headersMap.put("header1", "value1");
+    headersMap.put("header2", "value2");
+    credentialSourceMapWithUrlTextSource.put("headers", headersMap);
+    credentialSourceMapWithUrlJsonTextSource.put("headers", headersMap);
+
+    Map<String, String> textFormat = new HashMap<>();
+    textFormat.put("type", "text");
+
+    Map<String, String> jsonTextFormat = new HashMap<>();
+    jsonTextFormat.put("type", "json");
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java:437]: add uppercase letters? headers are case insensitive, but our code could be :)

**Review by TimurSadykov — CHANGES_REQUESTED** [review]
One small fix to make sure we support case insensitive headers

**Review by Neenu1995 — APPROVED** [review]

File: oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java [hunk 4]
```diff
@@ -48,10 +48,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java:51]: this has to stay as separate imports, maybe owl bot fixes that

File: oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java [hunk 5]
```diff
@@ -49,10 +49,7 @@
 import java.nio.file.Files;
 import java.nio.file.LinkOption;
 import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java:52]: also should keep individual imports

**Review by TimurSadykov — APPROVED** [review]
LGTM, but please remove "star" imports

**Pull request merged**